designs/auto-create-ecr.md

```diff
@@ -0,0 +1,306 @@
+Auto Create ECR Repos in Guided Deploy
+====================================
+
+
+What is the problem?
+--------------------
+
+With the release of Lambda Container Images Support in SAM CLI, customers today have to specify a ECR Repo URI location where images will need to be uploaded by SAM CLI after having been built. This means that customers need to have pre-created resources ready (in this case ECR repos) to go, so that they can supply them during the deploy process. This introduces friction and break in the seamless workflow that `sam deploy --guided` normally offers, with customers having to figure out how to create a ECR repo and how to find correct ECR URI to specify.
+
+Current flow for deploying a template with image based function:
+
+1. Create ECR repo: `aws ecr create-repository --repository-name myecr_repo`
+
+2. Deploy with SAM CLI guided: `sam deploy --guided`
+
+3. Specify image repo: `Image Repository for HelloWorldFunction []: 12345.dkr.ecr.us-west-2.amazonaws.com/helloworldfunctionrepo`
+
+What will be changed?
+---------------------
+
+When deploying with guided, SAM CLI will prompt the option to auto create ECR repos for image based functions.
+The auto created ECR repos will reside in a companion stack that gets deployed along with the actual stack. 
+
+During each guided deploy, the functions and repos will be synced. New repos will be created for new functions and repos without an associating function will be prompt for deletion.
+
+There will be an escape hatch to use non SAM CLI managed repos by specifying `--image-repositories` or change `samconfig.toml`.
```

Comment:
Would `--image-repository` also be an escape hatch or is that interface deprecated now?

Reply:
`--image-repository` can still be used as an escape hatch.